core/src/View/MDSpan/Kokkos_MDSpan_Extents.hpp

```diff
@@ -0,0 +1,156 @@
+// notice, this list of conditions and the following disclaimer in the
+// documentation and/or other materials provided with the distribution.
+//
+// 3. Neither the name of the Corporation nor the names of the
+// contributors may be used to endorse or promote products derived from
+// this software without specific prior written permission.
+//
+// THIS SOFTWARE IS PROVIDED BY NTESS "AS IS" AND ANY
+// EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT LIMITED TO, THE
+// IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS FOR A PARTICULAR
+// PURPOSE ARE DISCLAIMED. IN NO EVENT SHALL NTESS OR THE
+// CONTRIBUTORS BE LIABLE FOR ANY DIRECT, INDIRECT, INCIDENTAL, SPECIAL,
+// EXEMPLARY, OR CONSEQUENTIAL DAMAGES (INCLUDING, BUT NOT LIMITED TO,
+// PROCUREMENT OF SUBSTITUTE GOODS OR SERVICES; LOSS OF USE, DATA, OR
+// PROFITS; OR BUSINESS INTERRUPTION) HOWEVER CAUSED AND ON ANY THEORY OF
+// LIABILITY, WHETHER IN CONTRACT, STRICT LIABILITY, OR TORT (INCLUDING
+// NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE USE OF THIS
+// SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+//
+// Questions? Contact Christian R. Trott (crtrott@sandia.gov)
+//
+// ************************************************************************
+//@HEADER
+*/
+
+#ifndef KOKKOS_IMPL_PUBLIC_INCLUDE
+#include <Kokkos_Macros.hpp>
+#ifndef KOKKOS_ENABLE_DEPRECATED_CODE_3
+static_assert(false,
+              "Including non-public Kokkos header files is not allowed.");
+#else
+KOKKOS_IMPL_WARNING("Including non-public Kokkos header files is not allowed.")
+#endif
+#endif
+
+#ifndef KOKKOS_EXPERIMENTAL_MDSPAN_EXTENTS_HPP
+#define KOKKOS_EXPERIMENTAL_MDSPAN_EXTENTS_HPP
+
+#include <Kokkos_Core_fwd.hpp>
+
+#ifdef KOKKOS_ENABLE_IMPL_MDSPAN
+
+#include <experimental/mdspan>
+
+namespace Kokkos {
+namespace Impl {
+template <class DataType>
+struct ViewArrayAnalysis;
+
+template <std::size_t... Vals>
+struct ViewDimension;
+
+template <class T, class Dim>
+struct ViewDataType;
+}  // namespace Impl
+
+namespace Experimental::Impl {
+
+/*
+ * A few things to note --
+ * - mdspan allows for 0-rank extents similarly to View, so we don't need
+ * special handling of this case
+ * - View dynamic dimensions must be appear before static dimensions. This isn't
+ * a requirement in mdspan but won't cause an issue here
+ */
+
+template <std::size_t N>
+struct ExtentFromDimension {
+  static constexpr inline std::size_t value = N;
+};
+
+/*
+ * Kokkos uses a dimension of '0' to denote a dynamic dimension.
+ */
+template <>
+struct ExtentFromDimension<std::size_t{0}> {
+  static constexpr inline std::size_t value = std::experimental::dynamic_extent;
+};
+
+template <std::size_t N>
+struct DimensionFromExtent {
+  static constexpr inline std::size_t value = N;
+};
+
+template <>
+struct DimensionFromExtent<std::experimental::dynamic_extent> {
+  static constexpr inline std::size_t value = std::size_t{0};
+};
+
+template <class SizeType, class Dimension, class Indices>
```

Comment:
If we really want to leave control over it, then we better call it `SizeType -> IndexType`
```suggestion
template <class IndexType, class Dimension, class Indices>
```

Reply:
Removing it